presto-main/src/main/java/com/facebook/presto/execution/executor/TaskHandle.java

```diff
@@ -30,12 +30,12 @@
 import java.util.function.DoubleSupplier;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
-import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
 public class TaskHandle
 {
+    private volatile boolean destroyed;
```

Comment:
Why not use atomicboolean? Will be easier to assess correctness than volatile.

Reply:
For a couple reasons:
1. AtomicBoolean would have a (negligible) increase the total size of TaskHandle because of the extra object header and padding compared to a simple boolean, but would more importantly require an extra pointer dereference on each access. I moved the declaration of the field away from the mutable collections and to the beginning of the object layout as part of this change to attempt to avoid false sharing and cache misses in this PR for that reason.
2. It’s not immediately obvious to me that AtomicBoolean makes this flag any easier to reason about in this case compared to a simple volatile because there are no compare-and-set semantics in use. The flag is initially false and remains that way until it is set to true, after which it remains true (which is safe to permit multiple redundant calls to destroy()). With those semantics, AtomicBoolean get() and set() operations are equivalent to volatile reads and writes with the exception that they incur an extra indirection through the object reference which I preferred to avoid. However,  if you still disagree with that assessment and feel strongly about it, I’m happy to change it in another revision.

Reply:
My rule of thumb is to prefer simplicity over micro-optimization(unless we can prove the gains). AtomicBoolean is simpler than volatile because you don't have to think about false sharing. It'd work in any case.

However, this is a textbook usecase of volatile. `destroyed` will only be set once, so I don't mind if you prefer it like that. We have other uses of it in our codebase as well.